src/components/ProductList/useFiltering.js

```diff
@@ -0,0 +1,29 @@
+import {useState, useEffect} from 'react'
+
+const useFiltering = (data) => {
+    const [filters, setFilters] = useState([]);
+    const [filteredData, setFilteredData] = useState(data);
+
+    useEffect(() => {
+        if(filters.length > 0){
+            setFilteredData(data.filter(x => filters.includes(x.data.category.slug)));
+        } else {
+            setFilteredData(data);
+        }
+       
+    }, [filters, data]);
+
+    const handleCustomFilering = (newFilter) => {
```

Comment:
I think this filter is behaving  kind of weird. If you click one category it sets the filter it sets the style and appears as "visited" and you can click again to unset the filter, imo and from y experience the more intuitive thing you can do is, just treat them as if were part of a routing, you navigate between them and have an option to see "all", also maybe removing the class that sets a category as visited when you click another one.

Reply:
Sounds like a good idea, but it's not the behavior requested by the requirements. The expected behavior is the following: 
- 9.1. By default each category filter should be disabled.
- 9.2. When you click on the category filter for the first time, that filter will be enabled and applied to the product grid.
- 9.3. When the filter is active/enabled, you should apply some styling to the selected category so it’s easier to distinguish between active and inactive states.
- 9.4. You can have multiple filters enabled at the same time, so when this happens, the products that will be shown in the grid should have at least one of the categories from the active filters.
- 9.5. To disable a filter, you will have to click on it and it should be removed from the active filters.

Reply:
My bad, just a suggestion on how typically this kind of filters work. Maybe i'm suggesting that this requirement is changed a little bit to something more user intuitive